source/shared/cpp/ObjectModel/MarkDownHtmlGenerator.cpp

```diff
@@ -0,0 +1,198 @@
+#include "MarkDownHtmlGenerator.h"
+
+using namespace AdaptiveCards;
+
+std::string MarkDownStringHtmlGenerator::GenerateHtmlString()
+{
+    if (m_isHead) 
+    {              
+        m_token = (m_isHeadOfNewBlock)? "\n<p>" + m_token :"<p>" + m_token;
+    }
+
+    if (m_isTail)
+    {
+        return m_token + "</p>";
+    }
+
+    return m_token;
+}
+
+//     left and right emphasis tokens are match if
+//     1. they are same types
+//     2. neither of the emphasis tokens are both left and right emphasis tokens, and 
+//        if either or both of them are, then their sum is not multiple of 3 
+bool MarkDownEmphasisHtmlGenerator::IsMatch(std::shared_ptr<MarkDownEmphasisHtmlGenerator> &token)
+{
+    std::shared_ptr<MarkDownEmphasisHtmlGenerator> emphasisToken = 
+        std::static_pointer_cast<MarkDownEmphasisHtmlGenerator>(token);
```

Comment:
unnecessary cast

Reply:
thanks

---
In reply to: [157635640](https://github.com/Microsoft/AdaptiveCards/pull/1045#discussion_r157635640) [](ancestors = 157635640)